### PR TITLE
Spell Orthorhombic correctly; short names for point groups & reflection conditions

### LINKED ISSUE #21875: Orthorhomic misspelled
StatisticsOfPeaksWorkspaces should use short names for point groups. Also Orthorhomic is misspelled in the input (from the PointGroupFactory)

### PR DESCRIPTION
**Pull request by VickieLynch**
Description of work.
Fix mistake found in release by Xiaoping Wang:
Orthorhomic was misspelled in PointGroupFactory which populated options for StatisticsOfPeaksWorkspaces and SortHKL.  Spell correctly and add option for short names of point groups and reflections.

**To test:**
File is from UnitTest Data
```python
# Load example peak data and find cell
peaks = LoadIsawPeaks(Filename=r'Peaks5637.integrate')

FindUBUsingFFT(peaks, MinD=0.25, MaxD=10, Tolerance=0.2)
SelectCellWithForm(peaks, FormNumber=9, Apply=True, Tolerance=0.15)
OptimizeLatticeForCellType(peaks,
                           CellType='Hexagonal', Apply=True, Tolerance=0.2)

# Run the SortHKL algorithm
sorted, statistics_table = StatisticsOfPeaksWorkspace(peaks, PointGroup='-3m1 (Trigonal - Hexagonal)',
                                                      LatticeCentering='Rhombohedrally centred, obverse',
                                                      SortBy='Overall')
statistics = statistics_table.row(0)

peak = sorted.getPeak(0)
print "HKL of first peak in table %d" % peak.getH(),peak.getK(),peak.getL()
print "Multiplicity = %.2f" % statistics['Multiplicity']
print "Resolution Min = %.2f" % statistics['Resolution Min']
print "Resolution Max = %.2f" % statistics['Resolution Max']
print "No. of Unique Reflections = %i" % statistics['No. of Unique Reflections']
print "Mean ((I)/sd(I)) = %.2f" % statistics['Mean ((I)/sd(I))']
print "Rmerge = %.2f" % statistics['Rmerge']
print "Rpim = %.2f" % statistics['Rpim']
# Run again with short names
sorted, statistics_table = StatisticsOfPeaksWorkspace(peaks, PointGroup='-3m1',
                                                      LatticeCentering='Robv',
                                                      SortBy='Overall')
statistics = statistics_table.row(0)

peak = sorted.getPeak(0)
print "HKL of first peak in table %d" % peak.getH(),peak.getK(),peak.getL()
print "Multiplicity = %.2f" % statistics['Multiplicity']
print "Resolution Min = %.2f" % statistics['Resolution Min']
print "Resolution Max = %.2f" % statistics['Resolution Max']
print "No. of Unique Reflections = %i" % statistics['No. of Unique Reflections']
print "Mean ((I)/sd(I)) = %.2f" % statistics['Mean ((I)/sd(I))']
print "Rmerge = %.2f" % statistics['Rmerge']
print "Rpim = %.2f" % statistics['Rpim']
```

Fixes #21875 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@VickieLynch This fixes the problem.

The only issue I can see is if any scripts use one of the incorrectly named point groups at the moment then this will break after merging this. Is this worth dealing with do you think? We could do something similar to https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/mantid/simpleapi.py#L1034 and provide a deprecation warning?

File: Framework/Crystal/src/SortHKL.cpp [hunk 1]
```diff
@@ -45,15 +45,25 @@ void SortHKL::init() {
...
    * Probably there should be a dedicated Property type or validator. */
   std::vector<std::string> pgOptions;
-  pgOptions.reserve(m_pointGroups.size());
+  pgOptions.reserve(2 * m_pointGroups.size() + 5);
+  for (auto &pointGroup : m_pointGroups)
+    pgOptions.push_back(pointGroup->getSymbol());
   for (auto &pointGroup : m_pointGroups)
     pgOptions.push_back(pointGroup->getName());
+  // Scripts may have Orthorhombic misspelled from past bug in PointGroupFactory
+  pgOptions.push_back("222 (Orthorombic)");
```
Comment (martyngigg) [inline Framework/Crystal/src/SortHKL.cpp]: Thanks. I think we should check for the misspelling and log a warning. We can then remove them in a release or 2.

**Comment by VickieLynch** [comment]
Add this line to test warning
sorted, statistics_table = StatisticsOfPeaksWorkspace(peaks, PointGroup='222 (Orthorombic)',
                                                      LatticeCentering='Rhombohedrally centred, obverse',
                                                      SortBy='Overall')

**Review by martyngigg — APPROVED** [review]
Looks good. Thanks, @VickieLynch

**Comment by martyngigg** [comment]
The failing test here is unrelated to the changes.

**Pull request merged**